sensor/kubernetes/certrefresh/certificates/requester.go

```diff
@@ -187,37 +137,36 @@ func (r *genericRequester[ReqT, RespT]) RequestCertificates(ctx context.Context)
 		}
 	}
 
+	if r.ongoingRequest.Load() {
+		return nil, errors.New("concurrent requests are not supported.")
+	}
+	r.ongoingRequest.Store(true)
+	defer func() {
+		r.ongoingRequest.Store(false)
+	}()
+
 	requestID := uuid.NewV4().String()
-	receiveC := make(chan RespT, 1)
-	r.requests.Store(requestID, receiveC)
-	defer r.requests.Delete(requestID)
-	// Always delete this entry when leaving this scope to account for requests that are never responded, to avoid
-	// having entries in `r.requests` that are never removed.
+	r.ongoingRequestID = requestID
+	r.responseReceived.Reset()
+
 	if err := r.send(ctx, requestID); err != nil {
 		return nil, err
 	}
-	return r.receive(ctx, receiveC)
+	return r.receive(ctx)
 }
 
+// send a cert refresh request to Central
 func (r *genericRequester[ReqT, RespT]) send(ctx context.Context, requestID string) error {
-	// Assuming the `message.New` function is generic and can handle different request types.
 	msg := r.messageFactory.newMsgFromSensor(requestID)
-	select {
-	case <-r.stopC.Done():
-		return r.stopC.ErrorWithDefault(ErrCertificateRequesterStopped)
-	case <-ctx.Done():
-		return ctx.Err()
-	case r.sendC <- message.New(msg): // Use a generic `message.New` method for ReqT.
-		return nil
-	}
+	return r.msgToCentralFn(ctx, message.New(msg))
```

Comment:
(solution to the blocker) Now, if we have a local channel `messagesToCentral`, we could instead write to it here. Of course in a non-blocking fashion with `go func for select`.

Reply:
That would also require that we return that channel in some way - let's say, we have something like this:

```go
func (r *genericRequester[ReqT, RespT]) MessagesToCentral() <-chan *message.ExpiringMessage {
	return r. messagesToCentral
}
```